pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -72,10 +75,18 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	if err := c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner); err != nil {
-		return err
+	r.watchCheCluster = func() error {
+		return c.Watch(&source.Kind{Type: &orgv1.CheCluster{}}, enqueueRequestForOwner)
 	}
 
+	err = r.watchCheCluster()
+	if err != nil {
+		if _, ok := err.(*meta.NoKindMatchError); !ok { // ignore NoKindMatchError
+			return err
+		}
+	} else {
+		r.watchCheCluster = nil
+	}
```

Comment:
This is useful when `CheCluster` CRD already there and ctrl_init is executed.  So this is executed without error other than very first time.  This will make sure that the add_watch happen as early as possible and it has no dependency on `Reconcile()` when `CheCluster` CRD already available.